shop/src/controllers/products.ts

```diff
@@ -0,0 +1,75 @@
+import Products from "../models/Products";
+import { reqRes } from "../utils/interfaces";
+import { ProductNotFoundError } from "../utils/errors";
+import Product from "../models/Products";
+
+// CREATE
+
+export const postProduct: reqRes = async (req, res) => {
+    try {
+        const product = new Products(req.body)
+        await product.save()
+        res.status(201).send(product)
+    } catch (err) {
+        res.status(400).send("Server Error")
+    }
+}
+
+// READ ALL
+
+export const getProducts: reqRes = async (req, res) => {
+	try {
+		const products = await Products.find()
+		res.status(200).send(products)
+	} catch (err) {
+		res.status(500).send("Server Error")
+	}
+}
+
```

Comment:
I would have liked to see you attempt to handle the query params. David shared his version with a separate function, which is probably how it would be handled in a big company. This is the way I learned to handle query params. Basically just checking for query params in the getProducts call, and filtering the products if those query params exist. 

```export const getProducts: reqRes = async (req, res) => {
  try {
    let products = await Products.find();

    if (req.query.maxPrice) {
      const maxPrice = +req.query.maxPrice;
      products = products.filter((item) => item.price <= maxPrice);
    }

    if (req.query.includes) {
      const searchTerm = String(req.query.includes);
      products = products.filter((item) => item.name.includes(searchTerm));
    }

    if (req.query.limit) {
      const limit = +req.query.limit;
      products = products.slice(0, limit);
    }

    res.status(200).send(products);
  } catch (err) {
    res.status(500).send("Server Error");
  }
};```